app/src/androidTest/java/org/wikipedia/tests/SearchEngineTests.kt

```diff
@@ -0,0 +1,81 @@
+package org.wikipedia.tests
+
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import junit.framework.Assert.assertEquals
+import junit.framework.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.wikipedia.main.MainActivity
+import org.wikipedia.pages.*
+
+@RunWith(AndroidJUnit4::class)
+class SearchEngineTests {
+    val explorePage = ExplorePage()
+    val searchBarPage = SearchBarPage()
+    val searchPage = SearchPage()
+    val navBarPage = NavBarPage()
+    val onboardingPage = OnboardingPage()
+    val articlePage = ArticlePage()
+
+    @Before
+    fun beforeTests() {
+        onboardingPage.tapOnSkipButton()
+    }
+
+    @Rule
+    @JvmField
+    var mActivityTestRule = ActivityScenarioRule(MainActivity::class.java)
+
+
+    @Test
+    fun searchBarTest() {
+        var text = "Donald Trump"
+        explorePage.tapOnSearchBar()
+        searchBarPage.typeTextSearch(text)
+        assertTrue("No view matched '$text'", searchBarPage.isDisplayedElement())
```

Comment:
`searchBarPage.isDisplayedElement()` looks a bit off. This suggest that one should look at the search bar to see if there are any results. With POP, more sense would make if that check is performed on some sort of result list object. What is this check for?

Reply:
This checks if an element is displayed on the search result list. I changed the name of the method thus maybe it will be more clear what the test is exactly doing